airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/config/AirbyteConfiguredCatalog.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.config
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.airbyte.commons.json.Jsons
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog
+import io.micronaut.context.annotation.ConfigurationProperties
+import io.micronaut.core.util.StringUtils
+
+/**
+ * Micronaut configured properties holder for the Airbyte configured catalog provided to the
+ * connector CLI as an argument.
+ */
+@ConfigurationProperties("airbyte.connector.catalog")
+@SuppressFBWarnings(
+    value = ["NP_NONNULL_RETURN_VIOLATION"],
+    justification = "Uses dependency injection",
+)
+class AirbyteConfiguredCatalog {
+    lateinit var configured: String
+
+    fun getConfiguredCatalog(): ConfiguredAirbyteCatalog {
+        return if (StringUtils.isNotEmpty(configured)) {
+            Jsons.deserialize(
+                configured,
+                ConfiguredAirbyteCatalog::class.java,
+            )
+        } else {
+            ConfiguredAirbyteCatalog()
+        }
+    }
```

Comment:
Another DI newbie question, similar to the one I asked earlier about the enums: can we make the deserialization happen at instance construction time or thereabouts, instead of at access time? It's fine if we can't (or if we shouldn't obviously); I like to fail early fail fast, especially when failure is unlikely.

Reply:
We could also definitely do a lazy initialization on get to only deserialize it the first time it is used.

Reply:
Personally I'm most in favour of the custom setter alternative but I don't feel too strongly about this.